### ISSUE TITLE
smart extension breaks with lang=ru and xelatex

### ISSUE THREAD

**Issue by chebykinn** [jgm/pandoc#6192]
Hi, I've noticed that adding lang: ru breaks smart extension if I use xelatex pdf engine.

I'm able to reproduce this with this file:
```
---
lang: ru
header-includes:
  - \newfontfamily{\cyrillicfont}{Times New Roman} 
  - \newfontfamily{\cyrillicfontrm}{Times New Roman}
  - \newfontfamily{\cyrillicfonttt}{Courier New}
  - \newfontfamily{\cyrillicfontsf}{Arial}
---

# Test

Word -- some "definition".
```
build command: `pandoc bug.md --pdf-engine=xelatex -o bug.pdf`

With lang: ru it looks like this: 
![bug](https://user-images.githubusercontent.com/1508308/76794972-38feac80-67d9-11ea-85ba-d05c8241fd79.png)

Without it, everything looks fine:
![no-bug](https://user-images.githubusercontent.com/1508308/76794999-49168c00-67d9-11ea-835d-c146b94e0915.png)

pandoc version:
```
pandoc 2.9.1.1
Compiled with pandoc-types 1.20, texmath 0.12, skylighting 0.8.3.2
```
xelatex:
```
XeTeX 3.14159265-2.6-0.999991 (TeX Live 2019/Arch Linux)
kpathsea version 6.3.1
```

**Comment by jgm** [comment]
I guess the Russian polyglossia or babel redefines the ` `` ` ligature so it's no longer a curly quote.

**Comment by jgm** [comment]
LaTeX writer at line 316 has
``` haskell
    let ligatures = isEnabled Ext_smart opts && ctx == TextString
```
Maybe we should also disable ligatures when the language is Russian (and which others?)

**Comment by tarleb** [comment]
This was fixed in e26d31d56bc61553d5d6153f2130184643869877.
